python/whylogs/datasets/utils.py

```diff
@@ -0,0 +1,71 @@
+import os
+import re
+from datetime import date, datetime, timedelta
+from typing import Tuple, Union
+
+import pandas as pd
+
+from whylogs.datasets.configs import BaseConfig
+
+base_config = BaseConfig()
+
+
+def _adjust_date(current_date: str, date_offset: timedelta) -> datetime:
+    adjusted_date = _validate_timestamp(current_date) + date_offset
+    return adjusted_date
+
+
+def _parse_interval(interval: str) -> Tuple[int, str]:
+    try:
+        result = re.findall(r"^(\d+)([DdMmHhSs])", interval)[0]
+        return (int(result[0]), result[1].upper())
+    except IndexError:
+        raise ValueError("Could not parse interval!")
```

Comment:
good idea to add `f"Could not parse interval {interval}!"`

Reply:
Thanks!
Added this in the last commit!